bcda/cclf/fileprocessor_test.go

```diff
@@ -0,0 +1,221 @@
+	// assert that this file is still here.
+	_, err = os.Open(filePath)
+	assert.Nil(err)
+
+	testUtils.ResetFiles(s.Suite, BASE_FILE_PATH+"cclf/mixed/with_invalid_filenames/")
+
+	timeChange := origTime.Add(-(time.Hour * 73)).Truncate(time.Second)
+	err = os.Chtimes(filePath, timeChange, timeChange)
+	if err != nil {
+		s.FailNow("Failed to change modified time for file", err)
+	}
+
+	cclfMap, skipped, err = processCCLFArchives(folderPath)
+	assert.Nil(err)
+	cclfList = cclfMap["A0001"][18]
+	assert.Equal(2, len(cclfList))
+	assert.Equal(5, skipped)
+
+	// assert that this file is not still here.
+	_, err = os.Open(filePath)
+	assert.EqualError(err, "open ../../shared_files/cclf/mixed/with_invalid_filenames/T.BCDE.ACO.ZC0Y18.D181120.T0001000: no such file or directory")
+
+	testUtils.ResetFiles(s.Suite, BASE_FILE_PATH+"cclf/mixed/with_invalid_filenames/")
+}
+
+func (s *FileProcessorTestSuite) TestProcessCCLFArchives_InvalidPath() {
+	cclfMap, skipped, err := processCCLFArchives("./foo")
+	assert.EqualError(s.T(), err, "error in sorting cclf file: nil,: lstat ./foo: no such file or directory")
+	assert.Equal(s.T(), 0, skipped)
+	assert.Nil(s.T(), cclfMap)
+}
+
+func TestFileProcessorTestSuite(t *testing.T) {
+	suite.Run(t, new(FileProcessorTestSuite))
+}
+func TestGetCMSID(t *testing.T) {
+	tests := []struct {
+		name     string
+		path     string
+		hasError bool
+		cmsID    string
+	}{
+		{"validSSPPath", "path/T.BCD.A0001.ZCY18.D181120.T1000000", false, "A0001"},
+		{"validNGACOPath", "path/T.BCD.V299.ZCY19.D191005.T0209260", false, "V299"},
+		{"validCECPath", "path/T.BCD.E9999.ZCY19.D191005.T0209260", false, "E9999"},
+		{"missingBCD", "path/T.A0001.ACO.ZC8Y18.D18NOV20.T1000009", true, ""},
+		{"empty", "", true, ""},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(sub *testing.T) {
+			cmsID, err := getCMSID(tt.path)
+			if tt.hasError {
+				assert.Contains(sub, err.Error(), tt.path)
+			}
+			assert.Equal(sub, tt.cmsID, cmsID)
+		})
+	}
+}
+
+func TestGetCCLFMetadata(t *testing.T) {
+	cmsID := "A9999"
+	start := time.Now()
+	// Need to use UTC zone information to make the time comparison easier
+	// CCLF file format does not contain any tz information, so we assume UTC time
+	startUTC := time.Date(start.Year(), start.Month(), start.Day(), start.Hour(), start.Minute(), start.Second(), 0,
+		time.UTC)
+
+	gen := func(t time.Time) string {
+		const (
+			format         = "D060102.T1504050"
+			perfYearFormat = "06"
+		)
+		return fmt.Sprintf("T.BCD.A9999.ZC8Y%s.%s", t.Format(perfYearFormat), t.Format(format))
+	}
+
+	// Timestamp that'll satisfy the time window requirement
+	validTime := startUTC.Add(-24 * time.Hour)
+	validFile := gen(validTime)
+	validProdFile := strings.Replace(validFile, "T", "P", 1)
+
+	tests := []struct {
+		name     string
+		fileName string
+		errMsg   string
+		metadata cclfFileMetadata
+	}{
+		{"Non CCLF0 or CCLF8 file", "P.A0001.ACO.ZC9Y18.D190108.T2355000", "invalid filename", cclfFileMetadata{}},
```

Comment:
We might need some more tests here.  It would be nice to test the different flavors of files that come with CEC and NG.  For example, the CEC CCLF filenames look a little different, and I want to make sure our regex splitting works correctly.  For a filename like this `P.CEC.ZC020.D200908.T2355000`, what happens with the code in `getCCLFFileMetadata` since its looking for a regex split of size 6?